From 5baa1c43fa96750f239e7d756f98cc7a56e9cc01 Mon Sep 17 00:00:00 2001
From: "Song, Yoong Siang" <yoong.siang.song@intel.com>
Date: Mon, 27 Jul 2020 23:53:21 +0800
Subject: [PATCH 1/2] net: phy: Clear intr after enable link change intr for
 Marvell 88e2110 PHY

This commit clear intr status after enable the link change intr.

Signed-off-by: Song, Yoong Siang <yoong.siang.song@intel.com>
---
 drivers/net/phy/marvell10g.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/net/phy/marvell10g.c b/drivers/net/phy/marvell10g.c
index 385c806..6c0a4d2 100644
--- a/drivers/net/phy/marvell10g.c
+++ b/drivers/net/phy/marvell10g.c
@@ -675,6 +675,9 @@ static int m88e2110_set_wol(struct phy_device *phydev,
 
 		if (ret < 0)
 			return ret;
+
+		/* Clear the interrupt status register */
+		ret = phy_read_mmd(phydev, MDIO_MMD_PCS, MV_PCS_INTR_STS);
 	} else {
 		/* Disable the link status changed interrupt */
 		ret = phy_clear_bits_mmd(phydev, MDIO_MMD_PCS,
-- 
2.7.4

